docs/fsharp/language-reference/literals.md

```diff
@@ -39,6 +39,12 @@ The following table shows the literal types in F#. Characters that represent dig
 
 Values that are intended to be constants can be marked with the [Literal](https://fsharp.github.io/fsharp-core-docs/reference/fsharp-core-literalattribute.html) attribute. This attribute has the effect of causing a value to be compiled as a constant.
 
+Named literals are useful for:
+
+- Pattern matching without a `when` clause.
+- Attribute arguments.
+- Static type provider arguments.
```

Comment:
Links or examples of each would be useful here. However I'll merge without

Reply:
I do not have a quick visualization of the first here. I'm not sure what the code that uses that would look like, so an example would be awesome there. 

But I agree, lets merge this and then an additional fix with a sample of at least that one.